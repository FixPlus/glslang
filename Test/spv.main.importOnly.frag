#version 460 core

float foo(float, float);

void bar(inout vec3 v, int);
int bar(vec3);

float foo(int, float);

out float cout;

void main()
{
    vec3 v = vec3(0, 1, 2);
    bar(v, -1);
    int n = bar(v);
    float x = foo(1.f, 2.f);
    float y = foo(n, 2.f);
    cout = x + y;
}

float foo(float a, float b);
