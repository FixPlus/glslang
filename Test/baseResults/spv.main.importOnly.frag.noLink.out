spv.main.importOnly.frag
Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:12  Function Definition: main( ( global void)
0:12    Function Parameters: 
0:14    Sequence
0:14      Sequence
0:14        move second child to first child ( temp highp 3-component vector of float)
0:14          'v' ( temp highp 3-component vector of float)
0:14          Constant:
0:14            0.000000
0:14            1.000000
0:14            2.000000
0:15      Function Call: bar(vf3;i1; ( global void)
0:15        'v' ( temp highp 3-component vector of float)
0:15        Constant:
0:15          -1 (const int)
0:16      Sequence
0:16        move second child to first child ( temp highp int)
0:16          'n' ( temp highp int)
0:16          Function Call: bar(vf3; ( global highp int)
0:16            'v' ( temp highp 3-component vector of float)
0:17      Sequence
0:17        move second child to first child ( temp highp float)
0:17          'x' ( temp highp float)
0:17          Function Call: foo(f1;f1; ( global highp float)
0:17            Constant:
0:17              1.000000
0:17            Constant:
0:17              2.000000
0:18      Sequence
0:18        move second child to first child ( temp highp float)
0:18          'y' ( temp highp float)
0:18          Function Call: foo(i1;f1; ( global highp float)
0:18            'n' ( temp highp int)
0:18            Constant:
0:18              2.000000
0:19      move second child to first child ( temp highp float)
0:19        'cout' ( out highp float)
0:19        add ( temp highp float)
0:19          'x' ( temp highp float)
0:19          'y' ( temp highp float)
0:?   Linker Objects
0:?     Linker Function: foo(f1;f1; ( global highp float)
0:?       Function Parameters: 
0:?         '' ( in highp float)
0:?         '' ( in highp float)
0:?     Linker Function: bar(vf3;i1; ( global void)
0:?       Function Parameters: 
0:?         '' ( inout highp 3-component vector of float)
0:?         '' ( in highp int)
0:?     Linker Function: bar(vf3; ( global highp int)
0:?       Function Parameters: 
0:?         '' ( in highp 3-component vector of float)
0:?     Linker Function: foo(i1;f1; ( global highp float)
0:?       Function Parameters: 
0:?         '' ( in highp int)
0:?         '' ( in highp float)
0:?     'cout' ( out highp float)

Validation failed
// Module Version 10000
// Generated by (magic number): 80008
// Id's are bound by 56

                              Capability Shader
                              Capability Linkage
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 4  "main" 52
                              ExecutionMode 4 OriginUpperLeft
                              Source GLSL 460
                              Name 4  "main"
                              Name 11  "foo(f1;f1;"
                              Name 19  "bar(vf3;i1;"
                              Name 22  "bar(vf3;"
                              Name 26  "foo(i1;f1;"
                              Name 27  "v"
                              Name 33  "param"
                              Name 35  "param"
                              Name 38  "n"
                              Name 39  "param"
                              Name 42  "x"
                              Name 43  "param"
                              Name 44  "param"
                              Name 46  "y"
                              Name 47  "param"
                              Name 49  "param"
                              Name 52  "cout"
                              Decorate 11(foo(f1;f1;) Linkage Attributes 678391654 1715155302 15153 1
                              Decorate 19(bar(vf3;i1;) Linkage Attributes 678584674 993224310 3879273 1
                              Decorate 22(bar(vf3;) Linkage Attributes 678584674 993224310 0 1
                              Decorate 26(foo(i1;f1;) Linkage Attributes 678391654 1715155305 15153 1
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 32
               7:             TypePointer Function 6(float)
               8:             TypeFunction 6(float) 7(ptr) 7(ptr)
              12:             TypeVector 6(float) 3
              13:             TypePointer Function 12(fvec3)
              14:             TypeInt 32 1
              15:             TypePointer Function 14(int)
              16:             TypeFunction 2 13(ptr) 15(ptr)
              20:             TypeFunction 14(int) 13(ptr)
              23:             TypeFunction 6(float) 15(ptr) 7(ptr)
              28:    6(float) Constant 0
              29:    6(float) Constant 1065353216
              30:    6(float) Constant 1073741824
              31:   12(fvec3) ConstantComposite 28 29 30
              32:     14(int) Constant 4294967295
              51:             TypePointer Output 6(float)
        52(cout):     51(ptr) Variable Output
         4(main):           2 Function None 3
               5:             Label
           27(v):     13(ptr) Variable Function
       33(param):     13(ptr) Variable Function
       35(param):     15(ptr) Variable Function
           38(n):     15(ptr) Variable Function
       39(param):     13(ptr) Variable Function
           42(x):      7(ptr) Variable Function
       43(param):      7(ptr) Variable Function
       44(param):      7(ptr) Variable Function
           46(y):      7(ptr) Variable Function
       47(param):     15(ptr) Variable Function
       49(param):      7(ptr) Variable Function
                              Store 27(v) 31
              34:   12(fvec3) Load 27(v)
                              Store 33(param) 34
                              Store 35(param) 32
              36:           2 FunctionCall 19(bar(vf3;i1;) 33(param) 35(param)
              37:   12(fvec3) Load 33(param)
                              Store 27(v) 37
              40:   12(fvec3) Load 27(v)
                              Store 39(param) 40
              41:     14(int) FunctionCall 22(bar(vf3;) 39(param)
                              Store 38(n) 41
                              Store 43(param) 29
                              Store 44(param) 30
              45:    6(float) FunctionCall 11(foo(f1;f1;) 43(param) 44(param)
                              Store 42(x) 45
              48:     14(int) Load 38(n)
                              Store 47(param) 48
                              Store 49(param) 30
              50:    6(float) FunctionCall 26(foo(i1;f1;) 47(param) 49(param)
                              Store 46(y) 50
              53:    6(float) Load 42(x)
              54:    6(float) Load 46(y)
              55:    6(float) FAdd 53 54
                              Store 52(cout) 55
                              Return
                              FunctionEnd
  11(foo(f1;f1;):    6(float) Function None 8
               9:      7(ptr) FunctionParameter
              10:      7(ptr) FunctionParameter
                              FunctionEnd
 19(bar(vf3;i1;):           2 Function None 16
              17:     13(ptr) FunctionParameter
              18:     15(ptr) FunctionParameter
                              FunctionEnd
    22(bar(vf3;):     14(int) Function None 20
              21:     13(ptr) FunctionParameter
                              FunctionEnd
  26(foo(i1;f1;):    6(float) Function None 23
              24:     15(ptr) FunctionParameter
              25:      7(ptr) FunctionParameter
                              FunctionEnd
